lib/player.rb

```diff
@@ -0,0 +1,56 @@
+class Player
+    attr_reader   :name, :token, :cells_chosen
+  
+    def initialize(name,selected,board)
+      @board = board
+      @selected_token = selected
+      @name = welcome_player(name)
+      @token = get_token(@name)
+      @cells_chosen = []
+    end
+  
+    def add_cell(cell_number)
+      @cells_chosen.push(cell_number)
+    end
+
+    def welcome_player(player)
```

Comment:
this method might be unnecessary. It's simply running another method (get_name) and returning the result of that method call. Why not simply use get_name(player) in the line you used this?